### PR TITLE
 [xdoctest][task 365-370] reformat example code with google style in `paddle/incubate/nn/functional/` ,`paddle/incubate/optimizer/`

### PR DESCRIPTION
**Pull request by ooooo-create**
### PR types

Others

### PR changes

Others

### Description

修改如下文件的示例代码，使其通过 `xdoctest` 检查：

- `python/paddle/incubate/nn/functional/fused_layer_norm.py`
- `python/paddle/incubate/nn/functional/masked_multihead_attention.py`
- `python/paddle/incubate/nn/functional/variable_length_memory_efficient_attention.py`
- `python/paddle/incubate/optimizer/gradient_merge.py`
- `python/paddle/incubate/optimizer/lars_momentum.py`

预览：

- en: http://preview-paddle-pr-XXX.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/en/api/paddle/jit/XXX_en.html
- zh: http://preview-paddle-pr-XXX.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/jit/XXX_cn.html

### Related links
 
- #55629
- #55295

@sunzhongkai588 @SigureMo @megemini

### CODE REVIEW

File: python/paddle/incubate/optimizer/gradient_merge.py [hunk 1]
```diff
@@ -50,40 +50,41 @@ class GradientMergeOptimizer:
...
-        exe = base.Executor(place)
-        exe.run(base.default_startup_program())
-
-        for i in range(10):
-            cost_val = exe.run(feed=gen_data(32),
-                       program=base.default_main_program(),
-                       fetch_list=[cost.name])
-            print("step=%d, cost=%f" % (i, cost_val[0]))
+        >>> import paddle
+        >>> import paddle.base as base
```
Comment (SigureMo) [inline python/paddle/incubate/optimizer/gradient_merge.py]:
```suggestion
```

File: python/paddle/incubate/optimizer/gradient_merge.py [hunk 2]
```diff
@@ -50,40 +50,41 @@ class GradientMergeOptimizer:
...
+
+        >>> place = base.CPUPlace()
+        >>> exe = base.Executor(place)
+        >>> exe.run(base.default_startup_program())
+
+        >>> for i in range(10):
+        ...     cost_val = exe.run(feed=gen_data(32),
+        ...                program=base.default_main_program(),
+        ...                fetch_list=[cost.name])
+        ...     print("step=%d, cost=%f" % (i, cost_val[0]))
```
Comment (SigureMo) [inline python/paddle/incubate/optimizer/gradient_merge.py]:
```suggestion
        >>> place = paddle.CPUPlace()
        >>> exe = paddle.static.Executor(place)
        >>> exe.run(paddle.static.default_startup_program())

        >>> for i in range(10):
        ...     cost_val = exe.run(feed=gen_data(32),
        ...                program=paddle.static.default_main_program(),
        ...                fetch_list=[cost.name])
        ...     print("step=%d, cost=%f" % (i, cost_val[0]))
```

File: python/paddle/incubate/optimizer/lars_momentum.py [hunk 3]
```diff
@@ -63,24 +63,24 @@ class LarsMomentumOptimizer(Optimizer):
...
-            optimizer = base.optimizer.LarsMomentumOptimizer(learning_rate=0.001, momentum=0.9)
-            optimizer.minimize(out)
-
-            exe = base.Executor(base.CPUPlace())
-            exe.run(base.default_startup_program())
-            exe.run(
-                feed={"inp": np_inp},
-                fetch_list=[out.name])
+            >>> import paddle
+            >>> import paddle.base as base
```
Comment (SigureMo) [inline python/paddle/incubate/optimizer/lars_momentum.py]:
```suggestion
```

File: python/paddle/incubate/optimizer/lars_momentum.py [hunk 4]
```diff
@@ -63,24 +63,24 @@ class LarsMomentumOptimizer(Optimizer):
...
+            >>> np_inp = np.array([[1.0, 2.0], [3.0, 4.0]], dtype=np.float32)
+            >>> inp = paddle.static.data(
+            ...     name="inp", shape=[2, 2], dtype='float32')
+            >>> out = paddle.static.nn.fc(inp, size=3)
+            >>> out = paddle.sum(out)
+            >>> optimizer = paddle.incubate.optimizer.LarsMomentumOptimizer(learning_rate=0.001, momentum=0.9)
+            >>> optimizer.minimize(out)
+
+            >>> exe = base.Executor(base.CPUPlace())
+            >>> exe.run(base.default_startup_program())
```
Comment (SigureMo) [inline python/paddle/incubate/optimizer/lars_momentum.py]:
```suggestion
            >>> exe = paddle.static.Executor(base.CPUPlace())
            >>> exe.run(paddle.static.default_startup_program())
```

**Review by SigureMo — DISMISSED** [review]
LGTMeow 🐾

File: python/paddle/incubate/optimizer/lars_momentum.py [hunk 5]
```diff
@@ -63,24 +63,23 @@ class LarsMomentumOptimizer(Optimizer):
...
+            >>> paddle.enable_static()
+            >>> np_inp = np.array([[1.0, 2.0], [3.0, 4.0]], dtype=np.float32)
+            >>> inp = paddle.static.data(
+            ...     name="inp", shape=[2, 2], dtype='float32')
+            >>> out = paddle.static.nn.fc(inp, size=3)
+            >>> out = paddle.sum(out)
+            >>> optimizer = paddle.incubate.optimizer.LarsMomentumOptimizer(learning_rate=0.001, momentum=0.9)
+            >>> optimizer.minimize(out)
+
+            >>> exe = paddle.static.Executor(base.CPUPlace())
```
Comment (megemini) [inline python/paddle/incubate/optimizer/lars_momentum.py]: base.CPUPlace() -> paddle.CPUPlace()

Comment (SigureMo) [inline python/paddle/incubate/optimizer/lars_momentum.py]:
```suggestion
            >>> exe = paddle.static.Executor(paddle.CPUPlace())
```

**Review by SigureMo — APPROVED** [review]

**Review by megemini — APPROVED** [review]

**Pull request merged**